src/GraphQL.Analyzers.Tests/OneOfAnalyzerTests.cs

```diff
@@ -0,0 +1,245 @@
+
+        await VerifyCS.VerifyAnalyzerAsync(source, expected);
+    }
+
+    [Theory]
+    [InlineData(null)]
+    [InlineData("[OneOf]")]
+    [InlineData("[OneOfAttribute]")]
+    public async Task TypeFirst_NonNullableDiagnosticsReportedWhenOneOfAttribute(string? attribute)
+    {
+        string source =
+            $$"""
+              using GraphQL;
+
+              namespace Sample.Server;
+
+              {{attribute}}
+              public class MyInput
+              {
+                  public int? NullableProp { get; set; }
+                  public int? NullableField;
+
+                  public {|#0:int|} NonNullableProp { get; set; }
+                  public {|#1:int|} NonNullableField;
+
+              #nullable disable
+                  public string NullableRefProp1 { get; set; }
+                  public string NullableRefField1;
+
+              #nullable enable
+                  public string? NullableRefProp2 { get; set; }
+                  public string? NullableRefField2;
+
+                  public {|#2:string|} NonNullableRefProp { get; set; } = {|#3:null!|};
+                  public {|#4:string|} NonNullableRefField = {|#5:null!|};
+              }
+              """;
+
+        var expected = attribute != null
+            ?
+            [
+                VerifyCS.Diagnostic(OneOfAnalyzer.OneOfFieldsMustBeNullable).WithLocation(0),
+                VerifyCS.Diagnostic(OneOfAnalyzer.OneOfFieldsMustBeNullable).WithLocation(1),
+                VerifyCS.Diagnostic(OneOfAnalyzer.OneOfFieldsMustBeNullable).WithLocation(2),
+                VerifyCS.Diagnostic(OneOfAnalyzer.OneOfFieldsMustNotHaveDefaultValue).WithLocation(3),
+                VerifyCS.Diagnostic(OneOfAnalyzer.OneOfFieldsMustBeNullable).WithLocation(4),
+                VerifyCS.Diagnostic(OneOfAnalyzer.OneOfFieldsMustNotHaveDefaultValue).WithLocation(5)
+            ]
+            : DiagnosticResult.EmptyDiagnosticResults;
+
+        await VerifyCS.VerifyAnalyzerAsync(source, expected);
+    }
+
+    [Theory]
+    [InlineData(null)]
+    [InlineData("[OneOf]")]
+    [InlineData("[OneOfAttribute]")]
+    public async Task TypeFirst_DefaultValueDiagnosticsReportedWhenOneOfAttribute(string? attribute)
+    {
+        string source =
+            $$"""
+              using GraphQL;
+
+              namespace Sample.Server;
+
+              {{attribute}}
+              public class MyInput
+              {
+                  public int? NullableProp { get; set; } = {|#0:1|};
+                  public int? NullableField = {|#1:1|};
+
+              #nullable disable
+                  public string NullableRefProp1 { get; set; } = {|#2:"xxx"|};
+                  public string NullableRefField1 = {|#3:"xxx"|};
+
+              #nullable enable
+                  public string? NullableRefProp2 { get; set; } = {|#4:"yyy"|};
+                  public string? NullableRefField2 = {|#5:"yyy"|};
+              }
```

Comment:
Default values for properties defined in this manner are not copied to GraphQL (since reflection cannot read these values), and arguably should not cause a diagnostic.  Default values are defined as follows:

```cs
[DefaultValue(true)]
public bool Prop { get; set; }
```

Reply:
Oh, good to know. Will fix.